### DISCUSSION TITLE
Conditional Rules for Credit Card Field

### DISCUSSION THREAD

**Discussion by alexjcollins** [solspace/craft-freeform#163] [Questions & Answers] [answered]
I have a client looking to use payment features in a form, but the form may or may not require the submitter to pay.

To give some context it's a hardware fault report form and if the product is in warranty, the customer doesn't need to pay for the repair. If the product is out of warranty then the customer will need to pay for the repair.

I noticed the `Credit Card` field doesn't show in the conditional field rules (as far as I can tell!) - so does anyone know how I might achieve the conditional visibility and validation on the credit card field?

Cheers!

**Comment by kjmartens** [comment, marked as answer]
Currently, when using the Freeform Payments feature, the form MUST contain valid credit card information in order for the form to submit. For others that have experienced this need in the past, one workaround would be to have an intro page/area that lets the user decide (or choose a path) that sends them to a Payment form or no payment form. In most cases, this is used in the context of event signups, etc, and I can appreciate this might not work for your use-case.

If you're doing a simple "honor system" check on a date/year, you could have your intro page/form ask the user about the date they purchased it, and then direct them accordingly.

**Reply by alexjcollins** [reply]
Thanks, Kelsey! I thought that might be the case and a totally reasonable requirement. Like you suggest - I think running a few similar forms with a user-defined path makes the most sense.

Of course, it would then be handy if we could have reusable 'field groups' that allowed us to really easily drop in a selection of fields for reuse 😅

**Comment by ben-at-sahmri** [comment]
Sorry to re-open this long answered issue, but I'm hoping @kjmartens or someone else could give a little info on how to create the "path" choice mechanism described here. I'm assuming we have 3 freeform instances: one for the initial choice, and then one for each subsequent path. Is there a 'canonical free-form' way to do this? 

I imagine I want a way to route a form submission to a different URL based on the selections in the form - is that possible?

Thanks in advance!

**Reply by kjmartens** [reply]
I've responded to your other question here: #218 

To answer the question specifically here, you can check out the guide here:
**GUIDE:** [Success Return URL on Forms: **User Chooses Return URL**](https://docs.solspace.com/craft/freeform/v3/guides/success-return-on-forms.html#user-chooses-return-url)

🙂
